salt/states/boto_route53.py

```diff
@@ -360,53 +348,57 @@ def absent(
+def hosted_zone_present(name, domain_name=None, private_zone=False, caller_ref=None, comment='',
+                        vpc_id=None, vpc_name=None, vpc_region=None, region=None, key=None,
+                        keyid=None, profile=None):
     '''
-    Ensure a hosted zone exists with the given attributes.  Note that most
-    things cannot be modified once a zone is created - it must be deleted and
-    re-spun to update these attributes:
+    Ensure a hosted zone exists with the given attributes.  Note that most things cannot be
+    modified once a zone is created - it must be deleted and re-spun to update these attributes.
+    If you need the ability to update these attributes, please use the newer boto3_route53
+    module instead:
         - private_zone (AWS API limitation).
-        - comment (the appropriate call exists in the AWS API and in boto3, but
-                   has not, as of this writing, been added to boto2).
-        - vpc_id (same story - we really need to rewrite this module with boto3)
+        - comment (the appropriate call exists in the AWS API and in boto3, but has not, as of
+          this writing, been added to boto2).
+        - vpc_id (boto3 only)
         - vpc_name (really just a pointer to vpc_id anyway).
         - vpc_region (again, supported in boto3 but not boto2).
 
     name
-        The name of the state definition.  This will be used as the 'caller_ref'
-        param if/when creating the hosted zone.
+        The name of the state definition.
 
     domain_name
-        The name of the domain. This should be a fully-specified domain, and
-        should terminate with a period. This is the name you have registered
-        with your DNS registrar. It is also the name you will delegate from your
-        registrar to the Amazon Route 53 delegation servers returned in response
+        The name of the domain. This must be fully-qualified, terminating with a period.  This is
+        the name you have registered with your domain registrar.  It is also the name you will
+        delegate from your registrar to the Amazon Route 53 delegation servers returned in response
         to this request.  Defaults to the value of name if not provided.
 
-    comment
-        Any comments you want to include about the hosted zone.
-
     private_zone
         Set True if creating a private hosted zone.
 
+    caller_ref
+        A unique string that identifies the request and that allows create_hosted_zone() calls to be
+        retried without the risk of executing the operation twice.  This helps ensure idempotency
+        across state calls, but can cause issues if a zone is deleted and then an attempt is made
+        to recreate it with the same caller_ref.  If not provided, a unique UUID will be generated
+        at each state run, which avoids the risk of the above (transient) error.  This option is
+        generally not needed.  Maximum length of 128.
```

Comment:
Looks like you say what'll happen if it's not provided here, can you pull some of this copy into the state module as well?

Reply:
I've updated the exec module docstring to outline the pros and cons of passing (or not passing) caller_ref. :)